velox/type/Conversions.h

```diff
@@ -418,6 +455,51 @@ struct Converter<TypeKind::VARCHAR, void, TRUNCATE, LEGACY_CAST> {
   static std::string cast(const bool& val) {
     return val ? "true" : "false";
   }
+
+  static void normalizeStandardNotation(std::string& str) {
+    if (!FLAGS_experimental_enable_legacy_cast &&
+        str.find(".") == std::string::npos && isdigit(str[str.length() - 1])) {
+      str += ".0";
+    }
+  }
+
+  static void normalizeScientificNotation(std::string& str) {
+    auto idxE = str.find('E');
+    if (idxE == std::string::npos) {
+      return;
+    }
+
+    auto endCoef = idxE - 1;
+    while (endCoef > 0 && str[endCoef] == '0') {
+      --endCoef;
+    }
+    if (endCoef == 0) {
+      return;
+    }
+
+    auto pos = endCoef + 1;
+    if (str[endCoef] == '.') {
+      pos++;
+    }
+    str[pos++] = 'E';
+
+    auto startExp = idxE + 1;
+    if (str[startExp] == '-') {
+      str[pos++] = '-';
+      startExp++;
+    }
+    while (startExp < str.length() &&
+           (str[startExp] == '0' || str[startExp] == '+')) {
+      startExp++;
+    }
+    if (startExp == str.length()) {
+      return;
```

Comment:
We have done `str[pos++] = 'E'` above but haven't resize str. Would that mess up the result string if we return here?

Reply:
@kagamiori Wei, several boundary checks and returns in this func will never happen, given the returned string by fmt::format(":.16E"). I'm adding them just for completeness. Shall I remove them? Let me remove them, for perf.

Reply:
@gggrace14 For situations that should never happen, shall we add a VELOX_CHECK there instead of simply removing them? If performance is a concern, we can use VELOX_DCHECK.

Reply:
Hi @kagamiori , good idea! I can add VELOX_DCHECK. I think VELOX_CHECK is expensive if there's really error thrown.